src/lib/structures/expr.ml

```diff
@@ -1218,6 +1229,173 @@ let mk_builtin ~is_pos n l =
   if is_pos then pos else neg pos
 
 
+(** smart constructors for bit-vector terms *)
+
+let mk_bvconcat t1 t2 sz =
+  mk_term (Sy.Op Sy.Concat) [t1; t2] (Ty.Tbitv sz)
+
+let mk_bvextract i j t =
+  mk_term (Sy.Op (Sy.Extract (i, j))) [t] (Ty.Tbitv (j - i + 1))
+
+let mk_bvextend sign n k t =
+  if k = 0 then t else
+    mk_term (Op (BVExtend { sign; k })) [t] (Tbitv (n + k))
+
+let mk_bvnot n t =
+  mk_term (Sy.Op Sy.BVnot) [t] (Ty.Tbitv n)
+
+let mk_bvsign n t =
+  let t1 = mk_bvextract (n-1) (n-1) t in
+  mk_eq ~iff:false t1 bvone
+
+let mk_bvsign_pos n t =
+  mk_eq ~iff:false (mk_bvsign n t) bvone
+
+let mk_bvsign_neg n t =
+  mk_eq ~iff:false (mk_bvsign n t) bvzero
+
+let mk_bv2nat t =
+  mk_term (Sy.Op Sy.BV2Nat) [t] Ty.Tint
+
+let mk_nat2bv n t =
+  mk_term (Sy.Op (Sy.Nat2BV n)) [t] (Ty.Tbitv n)
+
+let mk_bvneg n x =
+  mk_nat2bv n (
+    mk_term (Sy.Op Sy.Minus) [
+      int Z.(pow ~$2 n |> to_string) ;
+      mk_bv2nat x
+    ] Ty.Tint
+  )
+
+let mk_bvult n x y =
+  if n = 0 then faux else
+    mk_builtin ~is_pos:true Sy.LT [ mk_bv2nat x; mk_bv2nat y ]
+
+let mk_bvule n x y =
+  if n = 0 then faux else
```

Comment:
I'd assume that `ule` is true for bv of size 0, not false?

Reply:
Intuitively yes, but, after testing, seems like both CVC5 and Z3 reject bit-vectors of size 0 at the parsing/typing phase, dunno if I missed it, but I don't remember reading this in the doc. Maybe dolmen need's to reject them too? (cc @Gbury)

Reply:
Seems like it is the case: https://smtlib.cs.uiowa.edu/theories-FixedSizeBitVectors.shtml
```
 :sorts_description "
    All sort symbols of the form (_ BitVec m)
    where m is a numeral greater than 0.
 "
```

Reply:
Indeed, it's bitvectors of size `0` should be refused. I'll open a PR in dolmen to fix that.

Reply:
In that case should the test `n = 0` be simply removed or be replaced by an assertion? Or we might want in some cases, internally to have 0-sized bit-vectors although I can't come up with a case in which we'd want that

Reply:
For now we should reject them when converting to Expr I'd say